pkg/fetch/git.go

```diff
@@ -80,9 +81,30 @@ func (t *Git) Retrieve(dstPath string) error {
 		env = append(env, "GIT_LFS_SKIP_SMUDGE=1")
 	}
 
+	gitUrl := t.opts.URL
+
+	gitCredentialsPath := filepath.Join(authDir.Path(), ".git-credentials")
+	if authOpts.Username != nil && authOpts.Password != nil {
+		if !strings.HasPrefix(gitUrl, "https://") {
+			return fmt.Errorf("username/password authentication is only supported for https remotes")
+		}
+		parsedUrl, err := url.Parse(gitUrl)
+		if err != nil {
+			return fmt.Errorf("could not parse git remote url: %w", err)
+		}
+		parsedUrl.User = url.UserPassword(*authOpts.Username, *authOpts.Password)
+		parsedUrl.Path = ""
+
+		err = ioutil.WriteFile(gitCredentialsPath, []byte(parsedUrl.String()+"\n"), 0600)
+		if err != nil {
+			return fmt.Errorf("writing .git-credentials: %s", err)
+		}
+	}
+
 	argss := [][]string{
 		{"init"},
-		{"remote", "add", "origin", t.opts.URL},
+		{"config", "credential.helper", "store --file " + gitCredentialsPath},
```

Comment:
just to confirm if this command runs in parallel with others git "setup" for other apps, it should be ok because it's scoped to the git directory vs being global?

Reply:
Yes, otherwise the `--global` flag would be needed.